packages/slice-machine/src/tracker.ts

```diff
@@ -0,0 +1,176 @@
+import type { Analytics as ClientAnalytics } from "@segment/analytics-next";
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { Frameworks } from "@slicemachine/core/build/src/models";
+import { LibraryUI } from "@models/common/LibraryUI";
+
+// These events should be sync with the tracking Plan on segment.
+type AllSliceMachineEventType = EventType | ContinueOnboardingType;
+
+enum EventType {
+  Review = "SliceMachine Review",
+  OnboardingStart = "SliceMachine Onboarding Start",
+  OnboardingSkip = "SliceMachine Onboarding Skip",
+  SlicePreviewSetup = "SliceMachine Slice Preview Setup",
+  SlicePreviewOpen = "SliceMachine Slice Preview Open",
+}
+
+export enum ContinueOnboardingType {
+  OnboardingContinueIntro = "SliceMachine Onboarding Continue Screen Intro",
+  OnboardingContinueScreen1 = "SliceMachine Onboarding Continue Screen 1",
+  OnboardingContinueScreen2 = "SliceMachine Onboarding Continue Screen 2",
+  OnboardingContinueScreen3 = "SliceMachine Onboarding Continue Screen 3",
+}
+
+let _client: ClientAnalytics | null = null;
```

Comment:
if you really want to keep a singleton, you can't let this client out there like this, it should be properly scoped.
That's kind of a standard pattern in js.
```
// the closure ensure that you scope your client inside and is accessible when calling public methods
const Tracker = (() => {
  let client

  // this part is actually the setup of the tracker itself
	const init = () => ({
		trackSomething() {
      console.log('track somethign here')
		}
	})

  // this part handles the singleton mecanism
	return () => ({
		get() {
			if (!client)  client = init();
      return client;
		}
	})
})();

// How it's supposed to be used
Tracker.get().trackSomething()
```

Closures enforce this.

Reply:
Done